plutus-tx-plugin/test/Strictness/let-default.eval-cek-catch.golden

```diff
@@ -1 +1,26 @@
-(con integer 1)
\ No newline at end of file
+(lam
```

Comment:
Wow, I'm absolutely shocked by this - WTF?! I have no explanation.

Reply:
Experimented with this - this can be corrected by inlining the arguments to `unsafeApplyCode` (as in https://github.com/input-output-hk/plutus/commit/dc372e5767623e8c8ea44fce8c249dc216c9e62d). No idea why, but this seems to be the case everywhere else in the codebase. Also see [slack thread](https://input-output-rnd.slack.com/archives/G01EE4NQ9U0/p1691185794036679).

Reply:
Ok I know what happened. Below is copied from the Slack thread:

With `Strict` on, the `CompiledCodeIn` type becomes strict. And because of `-fno-omit-interface-pragmas`, GHC is able to see through `plc`:

```haskell
plc _ _ = SerializedCode (mustBeReplaced "plc") (mustBeReplaced "pir") (mustBeReplaced "covidx")
```

and determines that `plc` simply returns bottom (which is not true - it will be rewritten by the plugin). GHC therefore simply ignores the second argument of `unsafeApplyCode`, and just returns the first argument! Lo and behold, the first argument is eventually re-written by the plugin, so it ends up being not bottom.

The solution is simple (which I verified): either make `CompiledCodeIn` lazy, or add `-fomit-interface-pragma` to the module that defines `plc`.

Reply:
Very impressive, great job!